python/ray/train/_internal/checkpoint_manager.py

```diff
@@ -99,6 +110,10 @@ def register_checkpoint(self, checkpoint_result: _TrainingResult):
                 key=self._get_checkpoint_score,
             )
         else:
+            if only_store_score_attr:
+                checkpoint_result = _TrainingResult(
+                    checkpoint=checkpoint_result.checkpoint, metrics={}
+                )
```

Comment:
Move the logic out of the if/else so that all the `only_store_score_attr` conditional logic is handled once in the beginning.

Reply:
Done.